cromwell_tools/cromwell_auth.py

```diff
@@ -136,32 +145,47 @@ def from_user_password(cls, username, password, url):
         auth = requests.auth.HTTPBasicAuth(username, password)
         return cls(url=url, header=None, auth=auth)
 
+    @classmethod
+    def from_no_authentication(cls, url):
+        """Generate an authentication object which does use any auth methods.
+
+        Args:
+            url (str): The URL to the Cromwell server. e.g. "https://cromwell.server.org/"
+
+        Returns:
+            CromwellAuth: An instance of this auth helper class with url and has no auth methods.
+        """
+        return cls(url=url, header=None, auth=None)
+
     @classmethod
     def harmonize_credentials(
             cls, username=None, password=None, url=None, secrets_file=None, service_account_key=None):
         """Parse and harmonize user inputted credentials and generate proper authentication object for cromwell-tools.
 
         Args:
-            username (str): Cromwell username for HTTPBasicAuth.
-            password (str): Cromwell password for HTTPBasicAuth.
-            url (str): The URL to the Cromwell server. e.g. "https://cromwell.server.org/"
-            secrets_file (str): Path to the JSON file containing username, password, and url fields. e.g.
+            username (str or None): Cromwell username for HTTPBasicAuth.
+            password (str or None): Cromwell password for HTTPBasicAuth.
+            url (str or None): The URL to the Cromwell server. e.g. "https://cromwell.server.org/"
+            secrets_file (str or None): Path to the JSON file containing username, password, and url fields. e.g.
                 {
                     "username": "",
                     "password": "",
                     "url": ""
                 }
-            service_account_key (str): Path to the JSON key file(service account key) for authenticating with CaaS.
+            service_account_key (str or None): Path to the JSON key file(service account key) for authenticating
+                with CaaS or any Cromwell instances that using OAuth.
 
         Returns:
             CromwellAuth: An instance of this auth helper class with proper credentials for authenticating with
                 Cromwell or CaaS.
         """
         # verify only one credential provided
         credentials = {
-            "service_account_key": True if service_account_key else False,
+            "service_account_key": True if all((service_account_key, url)) else False,
```

Comment:
Could this be simplified to just `all((service_account_key, url))`?

Reply:
It could, and in fact, it's better! Will do